ir/expression.def

```diff
@@ -66,10 +66,12 @@ class Sub : Operation_Binary {
 class Shl : Operation_Binary {
     stringOp = "<<";
     precedence = DBPrint::Prec_Shl;
+    Shl { if (type->is<Type::Unknown>() && left) type = left->type; }
```

Comment:
We want the types set for expressions in all cases, so it should be part of the constructor.  Making it a separate pass just leads to holes where the type info is incorrect.